_transcripts/032-rust-1.48-1.49.md

```diff
@@ -0,0 +1,731 @@
+
+__Jon__: Yeah, we got into, when we were discussing this before we started, we
+got into, like, what other kind of interesting things might be coming down the
+line. You want to talk a little bit about the slice functions you found?
+
+__Ben__: Yeah. Let me pull those up real quick. I believe they were called,
+`slice::array_chunks`. Is that correct?
+
+__Jon__: Yeah. It's like, `as_chunks` and `array_chunks`.
+
+__Ben__: There's also an `array_windows` function somewhere there too? Let's
+bring it up real quick.
+
+__Jon__: Yeah, there's, like, `as_chunks_mut`, `array_chunks_mut`,
+`array_windows`. And all of these are— they're pretty neat because they're
+basically a way to say, because I happen to have them open here. If you have a
+slice and you want to get, sort of, sub-slices of that slice, of different
+lengths. Like, you have a slice of length 8, and you want every— you want, like,
+slices of 2, chunks of 2, into that slice. There's now in nightly, there's this
+new method on slices called `as_chunks`, where it's generic over the length of
+the slice, and it returns you, like, sliced arrays, that each is one of those
+chunks. So if you want to slice into chunks of 2, what you're going to get back
+is a slice of array slices of length 2, that are those chunks. And it's just
+really neat, I think.
+
+__Ben__: Yeah, I think it's one of those really cool things that, once we start
+seeing it in the wild, and start internalizing the use of these const generics,
+it's going to be, I think, like, pretty idiomatic, and pretty widely
+distributed. So let's say, get used to it because I believe 1.51 is when
+`min_const_generics` is going to finally make it into stable. I believe it was
+stabilized a few days ago, as of this recording. So we should be having a lot of
+fun in the future.
+
+And we should probably talk about (_unintelligible— 10:41_) some crates like the
+vector math crates which I believe really want to use these to their full
+potential.
+
+__Jon__: Yeah. There were a couple of other APIs that were stabilized in this
+release as well. There's two in particular that I want to talk about, that are
+related to the futures ecosystem. So two functions that were stabilized are
+`future::ready` and `future::pending`, and these are kind of interesting. They
+are functions that you give them a— well, in the case of `future::ready`, you
+give it a type `T`, and it returns you a future of that type `T`, that just
+immediately yields `Ready` with the value you gave. And then `future::pending`
+is sort of the dual to that, in that you don't give it a value. You just give it
+a generic type `T`, and it will give you a future of that `T` that just never
+resolves. It always returns `Pending`.
+
+And you might wonder, like, why are these interesting? Like, why wouldn't I just
+use— at least for `future::ready`— why wouldn't I just use an async block, like
+an `async move t` because it gives you sort of the same functionality. And the
```

Comment:
I'm not sure how to format "async move t".  "async move || { t }" ?

Reply:
`async move { t }` is specifically what I was referring to